domain/src/Item.js

```diff
@@ -21,20 +21,39 @@ function calcAps(item) {
 
 function calcDps(item) {
     let res = (((item.minDmg + item.maxDmg) / 2) * item.aps);
-    if (item.critChance && item.critDmg) {
+    if (item.critChance > 0 && item.critDmg > 0) {
         if (item.critChance > 100) item.critChance = 100;
-        return (res + (res * (item.critChance / 100) * (item.critDmg / 100)));
+        res += (res * (item.critChance / 100) * (item.critDmg / 100));
     }
-    return res;
+    return res.toFixed(2);
 }
 
 function calcScore(item) {
     return item.dps * 10;
 }
 
+// > calcCharDps({dmg:10, atkSpd: 20})
+// 12
+function calcCharDps(char) {
+    let res = char.dmg + (char.dmg * (char.atkSpd / 100));
+    if (char.critChance > 0 && char.critDmg > 0) {
+        if (char.critChance > 100) char.critChance = 100;
+        res += (res * (char.critChance / 100) * (char.critDmg / 100));
+    }
+    return res.toFixed(2);
+}
+
+// > calcTotalDps({dps:20},{dmg:10, atkSpd: 20})
+// 32
+function calcTotalDps(item, char) {
+    return (item.dps * 1 + calcCharDps(char) * 1).toFixed(2);
```

Comment:
Strictly speaking, char don't have aps (attacks per second), but has ias (increase of attack speed). E.g. some other item can give +10% attack speed. It means that aps of a weapon will be increased by that amount. So if weapon has 1.0 aps without char modifier, it should have 1.0*(1+0.1)=1.1 aps with char modifier.

Same goes for damage. E.g. on ring you can have +150 dmg. If you weapon damage is 1500 without char modifier, it is 1500 + 150 = 1650 with char modifier.

So when you calculate total dps including all char modifiers, you need to add all those modifiers to corresponding stats and then use the usual formula.